### DISCUSSION TITLE
Update Problem from 1.15.3 to 2.0.0-beta-3

### DISCUSSION THREAD

**Discussion by jens-2305** [kimai/kimai#3811] [On-Premise: Installation & Updates]
Hello all,

at

 composer install --optimize-autoloader -n

I get the following error.

Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 255
!!  PHP Fatal error:  Declaration of App\Repository\SysadminRepository::upgradePassword(Symfony\Component\Security\Core\User\UserInterface $user, string $newEncodedPassword): void must be compatible with Symfony\Component\Security\Core\User\PasswordUpgraderInterface::upgradePassword(Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void in /var/www/html/kimai2/src/Repository/SysadminRepository.php on line 28
!!  Symfony\Component\ErrorHandler\Error\FatalError {#1479
!!    #message: "Compile Error: Declaration of App\Repository\SysadminRepository::upgradePassword(Symfony\Component\Security\Core\User\UserInterface $user, string $newEncodedPassword): void must be compatible with Symfony\Component\Security\Core\User\PasswordUpgraderInterface::upgradePassword(Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void"
!!    #code: 0
!!    #file: "./src/Repository/SysadminRepository.php"
!!    #line: 28
!!    -error: array:4 [
!!      "type" => 64
!!      "message" => "Declaration of App\Repository\SysadminRepository::upgradePassword(Symfony\Component\Security\Core\User\UserInterface $user, string $newEncodedPassword): void must be compatible with Symfony\Component\Security\Core\User\PasswordUpgraderInterface::upgradePassword(Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void"
!!      "file" => "/var/www/html/kimai2/src/Repository/SysadminRepository.php"
!!      "line" => 28
!!    ]
!!  }
!!

I have done the following
Update to php8.1

Then 
git checkout 2.0.0-beta-3
composer update symfony/flex --no-plugins --no-scripts

Deleted the existing plugins 

Can some one help

Thanks

**Comment by kevinpapst** [comment]
`SysadminRepository` is nothing from Kimai itself. Someone patched your local installation

**Reply by jens-2305** [reply]
I am the only one with access and have not edited any files. I only did the update

**Reply by kevinpapst** [reply]
That does not change the fact that `!! #file: "./src/Repository/SysadminRepository.php"` does not belong to Kimai

**Comment by jens-2305** [comment]
Wherever this file came from. I had to delete the following files and then it worked
/var/www/html/kimai2/src/Entity/Sysadmin.php
/var/www/html/kimai2/src/Repository/SysadminRepository.php

**Reply by kevinpapst** [reply]
Awesome 👍 Looking forward to read your beta-testing-feedback
